packages/core/strapi/lib/core/domain/component/index.js

```diff
@@ -1,22 +1,39 @@
 'use strict';
 
-const { cloneDeep, camelCase } = require('lodash/fp');
+const upperFirst = require('lodash/upperFirst');
+const cloneDeep = require('lodash/cloneDeep');
+const camelCase = require('lodash/camelCase');
+const startCase = require('lodash/startCase');
 const { validateComponentDefinition } = require('./validator');
 
-const createComponent = (definition = {}) => {
-  validateComponentDefinition(definition);
+/** @param {string} uid */
+function getCategoryFromUid(uid) {
+  let category = uid;
+  if (category.includes('::')) {
+    category = category.split('::')[1];
+  }
+  if (category.includes('.')) {
+    category = category.split('.')[0];
+  }
+  return startCase(category);
+}
 
-  const createdComponent = cloneDeep(definition);
-  const category = camelCase(definition.info.category);
+const createComponent = (uid, definition = {}) => {
+  try {
+    validateComponentDefinition(definition);
+  } catch (e) {
+    throw new Error(`Component Definition is invalid for ${uid}'.\n${e.errors}`);
+  }
 
-  const uid = `${category}.${definition.info.singularModelName}`;
+  const { schema } = cloneDeep(definition);
 
-  Object.assign(createdComponent, {
+  return Object.assign(schema, {
     uid,
-    category,
+    modelType: 'component',
+    modelName: schema.info.singularName,
+    category: getCategoryFromUid(uid),
```

Comment:
Here I may need some feedback/brainstorming. When later in related PR we want to load components from plugins, what would we use for the category? Plugin Name or leave the user to define the category in `schema.info.category`?